src/LEGO.AsyncAPI.Readers/V2/AsyncApiMessageDeserializer.cs

```diff
@@ -94,7 +96,13 @@ public static AsyncApiMessage LoadMessage(ParseNode node)
             var pointer = mapNode.GetReferencePointer();
             if (pointer != null)
             {
-                return mapNode.GetReferencedObject<AsyncApiMessage>(ReferenceType.Message, pointer);
+                // pointer is an internal components reference:
+                if (pointer.StartsWith("#")) return mapNode.GetReferencedObject<AsyncApiMessage>(ReferenceType.Message, pointer);
+
+                // how can we pass in settings and options for user to inject their own file reader
+                // is version matching important?
```

Comment:
Version matching becomes important once we need support for both V2 and V3 which is coming very soon. 
But important only on the read side, as we aren't looking to support writing to both version, but rather "upgrade" older versions to the newest. 

Does refs really allow for non-component refs? (External yes, but internal non-component refs?)